### PR TITLE
Sign up page and submission to api

### PR DESCRIPTION
**Pull request by camiriveron**
### Description
This PR creates the Sign up page UI and calls the API to create an user in the system. Since the react-redux-base project was used as a base, the changes to make sign up work are minor.

The Login Page will be defined in another User Story.

### Trello Reference

- [As a user I should be able to sign up so my account is created in Target MVD]( https://trello.com/c/0KVfVT0q/2-5-as-a-user-i-should-be-able-to-sign-up-so-my-account-is-created-in-target-mvd)

### Screenshots
For now we're just using an image as the Demo video placeholder:

screen shot 2018-11-13 at 12 44 51 pm

### CODE REVIEW

File: src/actions/sessionActions.js [hunk 1]
```diff
@@ -7,7 +7,7 @@ export const login = user =>
   async () => {
     try {
       const response = await sessionApi.login({ user });
```
Comment (kevquincke) [inline src/actions/sessionActions.js]: Maybe you can destructure the response object like so
```suggestion
      const { data } = await sessionApi.login({ user });
```

File: src/actions/sessionActions.js [hunk 2]
```diff
@@ -7,7 +7,7 @@ export const login = user =>
...
-      sessionService.saveUser(response.user);
+      sessionService.saveUser(response.data);
```
Comment (kevquincke) [inline src/actions/sessionActions.js]: And then use the data object
```suggestion
      sessionService.saveUser(data);
```

File: src/components/user/SignUpForm.js [hunk 3]
```diff
@@ -54,8 +66,19 @@ class SignUpForm extends PureComponent {
...
-          <FormattedMessage id="login.form.submit" />
+        <div>
+          <Field
+            name="gender"
+            label={intl.formatMessage(messages.gender)}
+            component="select"
+            className="select"
+          >
+            <option>{intl.formatMessage(messages.selectGender)}</option>
+            {gender.map((option, key) => <option key={key} value={option.value}>{option.text}</option>)}
```
Comment (kevquincke) [inline src/components/user/SignUpForm.js]: Same here
```suggestion
            {gender.map(({ value, text }, key) => <option key={key} value={value}>{text}</option>)}
```

File: src/containers/SignUpPage.js [hunk 4]
```diff
@@ -8,6 +8,8 @@ import { signUp } from 'actions/userActions';
 import SignUpForm from 'components/user/SignUpForm';
 import routes from 'constants/routesPaths';
 
+import demo from '../resources/images/demo3x.png';
```
Comment (kevquincke) [inline src/containers/SignUpPage.js]: Maybe you can add the resources folder to the resolved paths (you can find this file on `webpack/shared/resolve.js`) so you'll able to use absolute paths instead

**Review by kevquincke — COMMENTED** [review]
Good job!! 😄 I leave some minor comments. Also as a comment, you may find some people commenting on the styles to be ordered alphabetically, to be honest I prefer grouping them by functionality, but that's something debatable.

File: src/styles/styles.scss [hunk 5]
```diff
@@ -14,6 +22,160 @@ body {
...
+  align-items: center;
+  justify-content: center;
+  height: 100vh;
+}
+
+.demo__img {
+  display: block;
+  width: auto;
+  height: 100%;
+  object-fit: cover;
```
Comment (xlasserre) [inline src/styles/styles.scss]: Here you could also use the & from sass to organize the css, like so:
```
.demo {
     [styles]
      &__img {
          [styles for demo_img]
      }
}
```

**Review by xlasserre — APPROVED** [review]
Nice job!! 👏  Apart from kevin's comments I only added one more thing.

**Review by felipegarcia92 — APPROVED** [review]
Looking good! the other folks already commented

**Review by juanlacu — APPROVED** [review]
Good one! 👍

File: src/actions/sessionActions.js (hunk 2 above)
Comment (camiriveron) [reply src/actions/sessionActions.js]: Good call!

File: src/components/user/SignUpForm.js (hunk 3 above)
Comment (camiriveron) [reply src/components/user/SignUpForm.js]: I did't know you could do destructuring with a parameter like that, thanks!

**Review by kevquincke — APPROVED** [review]

**Pull request merged**